src/main/kotlin/com/openlattice/chronicle/services/candidates/CandidatesService.kt

```diff
@@ -0,0 +1,44 @@
+package com.openlattice.chronicle.services.candidates
+
+import com.openlattice.chronicle.candidates.Candidate
+import com.openlattice.chronicle.storage.ChroniclePostgresTables.Companion.CANDIDATES
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.EXPIRATION_DATE
+import com.openlattice.chronicle.storage.StorageResolver
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.sql.Connection
+
+@Service
+class CandidatesService(
+    private val storageResolver: StorageResolver,
+) : CandidatesManager {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(CandidatesService::class.java)
+
+        private val CANDIDATES_COLUMNS = CANDIDATES
+            .columns
+            .subtract(setOf(EXPIRATION_DATE))
+
+        private val INSERT_CANDIDATE_SQL = """
+            INSERT INTO ${CANDIDATES.name} (${CANDIDATES_COLUMNS.joinToString { it.name }}) 
+            VALUES (${CANDIDATES_COLUMNS.map { "?" }.joinToString()})
```

Comment:
Does this work without commas to separate the columns names and values?

Reply:
The default separator is `, `. This is what the query looks like:

```
INSERT INTO candidates (candidate_id, first_name, last_name, name, dob) 
VALUES (?, ?, ?, ?, ?)
```